build.sbt

```diff
@@ -1,31 +1,61 @@
-crossScalaVersions := Seq("2.13.6", "2.12.14")
-scalaVersion := crossScalaVersions.value.head
-
-name := "library-example"
+import xerial.sbt.Sonatype._
 
-// also used as a `groupId` by Sonatype
-organization := "ch.epfl.scala"
+lazy val `library-example` = project
+  .in(file("."))
+  .settings(
+    name := "library-example",
+    organization := "ch.epfl.scala",
+    description := "A library that does nothing useful",
+    crossScalaVersions := Seq("3.0.0", "2.13.6", "2.12.14"),
+    scalaVersion := crossScalaVersions.value.head,
+    libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test,
+    testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+  ).settings(publishSettings)
+   .settings(siteSettings)
+   .enablePlugins(MdocPlugin)
 
-libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val root = project
+  .in(file("."))
+  .aggregate(`library-example`)
```

Comment:
The `root` project is there only so that we can write `sbt test` instead of specifying library example.

I think most libraries consists of several modules and this build.sbt will be easier to replicate for them.
We can move `library-example` to its own directory and/or remove the top level `root`.

However I prefer "contained" build definitions rather than a flat build.sbt which becomes crowded with `ThisBuild /` as soon as there is module appearing.

Reply:
> We can move `library-example` to its own directory

OK for that.